File: Empresa/Proyecto2PHP/Entities/Cupon.php
<?php
class Cupon {
    public $idCupon;
    public $idCategoria;
    public $nombreUsuario;
    public $codigo;
    public $nombre;
    public $precio;
    public $descuento;
    public $ubicacion;
    public $imagenRepresentativa;
    public $fechaCreacion;
    public $fechaInicio;
    public $fechaFinalizacion;
    public $activo;

    public function __construct($idCupon, $idCategoria, $nombreUsuario, $codigo, $nombre, $precio, $descuento, $ubicacion, $imagenRepresentativa, $fechaCreacion, $fechaInicio, $fechaFinalizacion, $activo) {
        $this->idCupon = $idCupon;
        $this->idCategoria = $idCategoria;
        $this->nombreUsuario = $nombreUsuario;
        $this->codigo = $codigo;
        $this->nombre = $nombre;
        $this->precio = $precio;
        $this->descuento = $descuento;
        $this->ubicacion = $ubicacion;
        $this->imagenRepresentativa = $imagenRepresentativa;
        $this->fechaCreacion = $fechaCreacion;
        $this->fechaInicio = $fechaInicio;
        $this->fechaFinalizacion = $fechaFinalizacion;
        $this->activo = $activo;
    }
}
?>

File: Cliente/Proyecto2Cliente/Proyecto2.BC/Modelos/Cupon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proyecto2.BC.Modelos
{
    public class Cupon
    {
        public int idCupon { get; set; }
        public int idCategoria { get; set; }

        public string? nombreUsuario { get; set; }
        public string? nombreCategoria { get; set; }
        public string codigo { get; set; } = null!;
        public string nombre { get; set; } = null!;
        public string? nombreEmpresa { get; set; }
        public double precio { get; set; }
        public double? descuento { get; set; }
        public string? ubicacion { get; set; }
        public string? imagenRepresentativa { get; set; }
        public DateTime fechaCreacion { get; set; }
        public DateTime fechaInicio { get; set; }
        public DateTime fechaFinalizacion { get; set; }
        public bool activo { get; set; }
    }
}
